goalgorilla/open_social
# Feature/3056821 ajaxify notification centre

## Problem
The notification centre loads a view on every page, while it's not immediately visible to the user. Therefor the page load performance drops significantly.

## Solution
1. Ajaxified the notification view to render via controller callback.
2. Added a **load_link** element in `social_activity_social_user_account_header_items`
3. Made changes in notification.js to click the link to trigger ajax callback and show the notifications
4. Added cache max-age 0 to prevent from caching.

## Issue tracker
https://www.drupal.org/project/social/issues/3056821

## How to test
- Login via two users
- Try to create notifications for any of the user.
- Click the notification bell icon
- It should load via notifications via ajax
- Hide/show should work
- Notifications should be mark as seen
- Subsequent page load should not load the render html from activity notification view until notification bell icon is clicked (Check in developer tools for present of view's html)

## Release notes
Deprecated `social/modules/custom/activity_creator/src/Controller/NotificationsController.php::readNotificationCallback()`
